src/core/hle/service/http/http_c.cpp

```diff
@@ -48,53 +52,138 @@ const ResultCode ERROR_NOT_IMPLEMENTED = // 0xD960A3F4
+    ErrCodes::HeaderNotFound, ErrorModule::HTTP, ErrorSummary::InvalidState, ErrorLevel::Permanent);
+const ResultCode ERROR_BUFFER_SMALL = ResultCode(ErrCodes::BufferTooSmall, ErrorModule::HTTP,
+                                                 ErrorSummary::WouldBlock, ErrorLevel::Permanent);
 const ResultCode ERROR_WRONG_CERT_ID = // 0xD8E0B839
     ResultCode(57, ErrorModule::SSL, ErrorSummary::InvalidArgument, ErrorLevel::Permanent);
 const ResultCode ERROR_WRONG_CERT_HANDLE = // 0xD8A0A0C9
     ResultCode(201, ErrorModule::HTTP, ErrorSummary::InvalidState, ErrorLevel::Permanent);
 const ResultCode ERROR_CERT_ALREADY_SET = // 0xD8A0A03D
     ResultCode(61, ErrorModule::HTTP, ErrorSummary::InvalidState, ErrorLevel::Permanent);
 
-static std::pair<std::string, std::string> SplitUrl(const std::string& url) {
+struct URLInfo {
+    bool is_https;
+    std::string host;
+    int port;
+    std::string path;
+};
+
+// Splits URL into its components. Example: https://citra-emu.org:443/index.html
+// is_https: true; host: citra-emu.org; port: 443; path: /index.html
+static URLInfo SplitUrl(const std::string& url) {
     const std::string prefix = "://";
     const auto scheme_end = url.find(prefix);
     const auto prefix_end = scheme_end == std::string::npos ? 0 : scheme_end + prefix.length();
 
+    bool is_https = scheme_end != std::string::npos && url.starts_with("https");
+
     const auto path_index = url.find("/", prefix_end);
     std::string host;
+    int port = -1;
+    constexpr int default_http_port = 80;
+    constexpr int default_https_port = 443;
     std::string path;
     if (path_index == std::string::npos) {
         // If no path is specified after the host, set it to "/"
-        host = url;
+        host = url.substr(prefix_end);
+        const auto port_start = host.find(":");
+        if (port_start != std::string::npos) {
+            std::string port_str = host.substr(port_start + 1);
+            host = host.substr(0, port_start);
+            char* pEnd = NULL;
+            port = std::strtol(port_str.c_str(), &pEnd, 10);
+            if (*pEnd) {
+                port = -1;
+            }
+        }
         path = "/";
     } else {
-        host = url.substr(0, path_index);
+        host = url.substr(prefix_end, path_index - prefix_end);
+        const auto port_start = host.find(":");
+        if (port_start != std::string::npos) {
+            std::string port_str = host.substr(port_start + 1);
+            host = host.substr(0, port_start);
+            char* pEnd = NULL;
+            port = std::strtol(port_str.c_str(), &pEnd, 10);
+            if (*pEnd) {
+                port = -1;
+            }
+        }
         path = url.substr(path_index);
     }
-    return std::make_pair(host, path);
+    if (port == -1) {
+        port = is_https ? default_https_port : default_http_port;
+    }
+    return URLInfo{.is_https = is_https, .host = host, .port = port, .path = path};
```

Comment:
```suggestion
    return URLInfo{.is_https = is_https, .host = host, .port = port, .path = path,};
```
This will force clang format to format it properly